File: CPP/No274.cc
/**
 * Created by Xiaozhong on 2020/8/24.
 * Copyright (c) 2020/8/24 Xiaozhong. All rights reserved.
 */
#include "vector"
#include "algorithm"

using namespace std;

class Solution {
public:
    int hIndex(vector<int> &citations) {
        sort(citations.begin(), citations.end());
        int i = 0;
        while (i < citations.size() && citations[citations.size() - 1 - i] > i) i++;
        return i;
    }
};

File: Java/src/No274.java
/*
 * Copyright (c) 7/11/2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

import java.util.Arrays;

public class No274 {
    public int hIndex(int[] citations) {
        int n = citations.length;
        Integer[] nums = new Integer[n];
        for (int i = 0; i < n; ++i) nums[i] = citations[i];
        Arrays.sort(nums, (o1, o2) -> o2 - o1);
        for (int i = 0; i < n; ++i) {
            if (nums[i] < i + 1) return i;
        }
        return n;
    }

    public static void main(String[] args) {
        int[] citations = {1, 3, 1};
        No274 no274 = new No274();
        System.out.println(no274.hIndex(citations));
    }
}
